tests/4.5/task/test_task_critical.c

```diff
@@ -0,0 +1,119 @@
+//===-- test_task_critical.c ------------------------------------------------===//
+//
+// OpenMP API Version 4.5 Nov 2015
+//
+// Description:
+// testTaskWithCriticalBasic()
+// This is a basic test to demonstrate how a shared resource
+// can be accessed and written to in multiple thread environment.
+//
+//testTaskWithCriticalAdvanced()
+// This is a advanced test to demonstrate how a shared resource
+// can be accessed and written to in multiple thread environment.
+// In this a buffer is filled with random integers. The parallel
+// region counts the number of prime integers.
+//===----------------------------------------------------------------------===//
+
+
+
+#include <stdio.h>
+#include <assert.h>
+#include <stdlib.h>
+#include <omp.h>
+#include "ompvv.h"
+
+int testTaskWithCriticalBasic(int NThrds) {
+  int errors = 0;
+  int count = 0;
+  omp_set_num_threads(NThrds);
+#pragma omp parallel
+  {
```

Comment:
NThreads may not be what you get.

Reply:
Avoided passing arbitrary number of threads.